### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Saverio976**

### CODE REVIEW

File: include/Arcade/ECS/IEventManager.hpp [hunk 1]
```diff
@@ -109,17 +109,23 @@ namespace Arcade {
...
-                 * bool (True if the event was trigered, False if the event was not trigered)
-                 * and std::optional<std::shared_ptr<IComponent>> (The parameter passed as parameter to `addEvent` method)
+                 * bool (True if the event was trigered, False if the event was
+                 * not trigered) and std::optional<std::shared_ptr<IComponent>>
+                 * (The parameter passed as parameter to `addEvent` method)
                  */
-                virtual std::pair<bool, std::optional<std::shared_ptr<IComponent>>> isEventInQueue(const std::string &event) const = 0;
+                virtual std::pair<bool,
+                std::optional<std::shared_ptr<IComponent>>>
+                isEventInQueue(const std::string &event) const = 0;
```
Comment (guillaumeAbel) [inline include/Arcade/ECS/IEventManager.hpp:118]: We don't implement Queue but Vectors

File: include/Arcade/Graph/ISprite.hpp [hunk 2]
```diff
@@ -72,25 +78,41 @@ namespace Arcade {
...
-                virtual Rect &getRect() const = 0;
+                virtual const Rect &getRect() const = 0;
+                /**
+                 * @brief Set the sprite rectangle.
+                 */
+                virtual void setRect(Rect &rect) = 0;
+                /**
+                 * @brief Set the sprite rectangle.
+                 */
+                virtual void setRect(const Rect &rect) = 0;
```
Comment (Saverio976) [inline include/Arcade/Graph/ISprite.hpp:105]: 2 setRect ?

File: include/Arcade/Core/IGameModule.hpp [hunk 3]
```diff
@@ -37,6 +38,14 @@ namespace Arcade {
                  *
                  */
                 virtual void changeGame(const std::string &gameName) = 0;
+
+                /**
+                 * @brief Change the current active game to the next
```
[the remark this answers is not in the document]
Comment (guillaumeAbel) [reply include/Arcade/Core/IGameModule.hpp:43]: we don't go to next game in lib folder but in GameModule games lib's name list, but it's many to precize and its can be different according implementation

[the remark this answers is not in the document]
Comment (Saverio976) [reply include/Arcade/Core/IGameModule.hpp:43]: can you add to doc that it need to go to next (in a cyclic way)

**Pull request merged**